### PR TITLE
Integrate Analytics

### PR DESCRIPTION
**Pull request by mjangda**
With the analytics lib introduced in #118, we can now actually integrate it to track things.

- [x] Integrate with login
- [x] Integrate with command framework (e.g. validation, help, context, etc.)
- [x] Integrate with `vip app`
- [x] Integrate with `vip app list`
- [x] Integrate with `vip sync`
- [ ] Add disclaimer on login
- [ ] Add opt-out
- [x] Make sure analytics failures are non-critical
- [x] Review all events to make sure they are necessary and useful
- [ ] Switch uuid to GH ID after login.

### CODE REVIEW

File: src/lib/cli/command.js [hunk 1]
```diff
@@ -29,21 +28,17 @@ function uncaughtError( err ) {
...
-	const tracker = await initAnalytics();
-
 	const options = this.parse( argv );
 
 	const validationError = validateOpts( options );
 	if ( validationError ) {
-		console.log( validationError.toString() );
+		const error = validationError.toString();
+
+		trackEvent( 'command_validation_error', { error } );
```
Comment (joshbetz) [inline src/lib/cli/command.js]: Is there any risk of process.exit() firing before this Promise resolves?

**Comment by joshbetz** [comment]
Overall looks good. There are some instances where I'm not sure if `trackEvent` will resolve before the CLI exits. The idea of "fire and forget" is nice, but I'm not sure how it works in practice since we're not working with a long-running process.

File: src/lib/cli/command.js (hunk 1 above)
Comment (mjangda) [reply src/lib/cli/command.js]: Oh, good catch. We need `await` before all those calls.

**Comment by mjangda** [comment]
> The idea of "fire and forget" is nice, but I'm not sure how it works in practice since we're not working with a long-running process.

Those are all supposed to be blocking async actions (which is now fixed).

I'm not totally happy about the implementation (both the blocking aspect and the extra complexity it adds). It would be nice to move towards a single request instead (maybe firing on `beforeExit` + `uncaughtException`) and a cleaner implementation (maybe using events?).

**Comment by mjangda** [comment]
Going to move the the disclaimer, opt-out, and Tracks user alias to new Issues + PRs.

**Pull request merged**